ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -974,6 +975,94 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
     return documents
 
 
+@with_traced_module
+def traced_base_tool_invoke(langchain, pin, func, instance, args, kwargs):
+    integration = langchain._datadog_integration
+    tool_input = get_argument_value(args, kwargs, 0, "input")
+    config = get_argument_value(args, kwargs, 1, "config", optional=True)
+
+    span = integration.trace(
+        pin,
+        "%s.%s.%s.%s" % (func.__module__, func.__class__.__name__, func.__name__, func.__self__.name),
```

Comment:
I think this might be overkill on the span resource naming. Ideally we only need to know the name of the tool being invoked.
IMO it's unnecessary to add `method.invoke` as part of the span name, the tool name itself should make it fairly obvious that this is a tool invocation. Same with the underlying module, not sure if this gives users much helpful information that they don't already have.
Can we get just the tool's name as the resource? i.e. `Circumference calculator`

Reply:
My bad I forgot i had the invoke and ainvoke and only updated one of them 🦆